pipelines/controller/src/main/java/org/openmrs/analytics/ThriftServerHiveResourceManager.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openmrs.analytics;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** This class manages to create resources on Thrift Server post each pipeline run. */
+public class ThriftServerHiveResourceManager {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(ThriftServerHiveResourceManager.class.getName());
+  private final String jdbcDriverClass;
+  private final String jdbcUrl;
+  private final String user;
+  private final String password;
+
+  public ThriftServerHiveResourceManager(
+      String jdbcDriverClass, String jdbcUrl, String user, String password) {
+    this.jdbcDriverClass = jdbcDriverClass;
+    this.jdbcUrl = jdbcUrl;
+    this.user = user;
+    this.password = password;
+  }
+
+  /**
+   * Method to create resources on Thrift Server Hive.
+   *
+   * @param resourceList Comma separated list of resources such as Patient, Observation, Encounter
+   * @param timestamp Timestamp suffix to be used in table name.
+   * @param thriftServerParquetPath location of parquet files in Thrift Server
+   * @throws SQLException
+   */
+  public void createResources(String resourceList, String timestamp, String thriftServerParquetPath)
+      throws SQLException {
+    try {
+      Class.forName(jdbcDriverClass);
```

Comment:
Are we passing `jdbcDriverClass` to this class only to do this check? If yes, can we remove it from here and instead do this validation in `DataProperties.validateProperties` instead? Beside making this class simpler, that is a better place for doing this check and maybe even fail early (at the very beginning) instead of running the pipelines and just log an error at the end.

Reply:
ack